client/extensions/wp-job-manager/components/setup/index.jsx

```diff
@@ -16,41 +16,70 @@ import Intro from './intro';
 import Main from 'components/main';
 import PageSetup from './page-setup';
 import Wizard from 'components/wizard';
-import { getSelectedSiteSlug } from 'state/ui/selectors';
-
-const SetupWizard = ( {
-	slug,
-	stepName = Steps.INTRO,
-	translate,
-} ) => {
-	const steps = [ Steps.INTRO, Steps.PAGE_SETUP, Steps.CONFIRMATION ];
-	const components = {
-		[ Steps.INTRO ]: <Intro />,
-		[ Steps.PAGE_SETUP ]: <PageSetup />,
-		[ Steps.CONFIRMATION ]: <Confirmation />,
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { saveSetupStatus } from '../../state/setup/actions';
+
+class SetupWizard extends Component {
+	static propTypes = {
+		saveSetupStatus: PropTypes.func.isRequired,
+		siteId: PropTypes.number,
+		slug: PropTypes.string,
+		stepName: PropTypes.string,
+		translate: PropTypes.func.isRequired,
+	};
+
+	static defaultProps = {
+		stepName: Steps.INTRO,
 	};
-	const mainClassName = 'wp-job-manager__setup';
-
-	return (
-		<Main className={ mainClassName }>
-			<DocumentHead title={ translate( 'Setup' ) } />
-			<Wizard
-				basePath={ `${ SetupPath }/${ slug }` }
-				components={ components }
-				forwardText={ translate( 'Continue' ) }
-				hideNavigation={ true }
-				steps={ steps }
-				stepName={ stepName } />
-		</Main>
-	);
-};
-
-const mapStateToProps = state => ( { slug: getSelectedSiteSlug( state ) } );
-
-SetupWizard.propTypes = {
-	slug: PropTypes.string,
-	stepName: PropTypes.string,
-	translate: PropTypes.func.isRequired,
-};
-
-export default connect( mapStateToProps )( localize( SetupWizard ) );
+
+	componentDidMount() {
+		const { siteId } = this.props;
+
+		if ( ! siteId ) {
+			return;
+		}
+
+		this.props.saveSetupStatus( siteId, false );
```

Comment:
Maybe we could add a brief comment here (and below, in `componentWillReceiveProps`) so people looking at the code later (without the benefit of your really nice PR description) can understand why it makes sense to set the status to `false` at these points?